src/Agent.cs

```diff
@@ -26,6 +26,14 @@ public class Agent
         private readonly Collector _collector;
         private static readonly ILog _log = LogManager.GetCurrentClassLogger();
 
+        public int MessageCount
+        {
+            get
+            {
+                return _collector.MessageCount;
```

Comment:
That is to say, I added this change to #11, but in the collector class, rather than here.